qiskit/algorithms/phase_estimators/phase_estimation.py

```diff
@@ -208,10 +247,10 @@ def estimate_from_pe_circuit(
     # pylint: disable=missing-param-doc
     def estimate(
         self,
-        unitary: Optional[QuantumCircuit] = None,
-        state_preparation: Optional[QuantumCircuit] = None,
-        pe_circuit: Optional[QuantumCircuit] = None,
-        num_unitary_qubits: Optional[int] = None,
+        unitary: QuantumCircuit | None = None,
+        state_preparation: QuantumCircuit | None = None,
+        pe_circuit: QuantumCircuit | None = None,
```

Comment:
It seems from below that `pe_circuit` use has been deprecated since 0.18. Since we are long past the date when this could have been addressed (see deprecation message below) I think we should take care of this now with this refactoring.